main.py

```diff
@@ -971,6 +1053,72 @@ def suggest(limit=10):
     return better_jsonify(suggestions=suggestions)
 
 
+@app.route("/locations", methods=["GET"])
+@max_age(seconds=60 * 5)
+def locations():
+    """ Render locations page """
+    kind = request.args.get("kind")
+    period = request.args.get("period")
+
+    days = _TOP_LOCATIONS_PERIOD
+    if period == "week":
+        days = 7
+
+    kind = kind if kind in LOCATION_TAXONOMY else None
+
+    with SessionContext(commit=False) as session:
+        locs = top_locations(enclosing_session=session, kind=kind, days=days)
+
+    return render_template("locations/locations.html", locations=locs, period=period)
+
+
+@app.route("/staticmap", methods=["GET"])
+@cache.cached(timeout=60 * 60 * 24, key_prefix="staticmap", query_string=True)
+def staticmap():
+    """ Proxy for Google Static Maps API """
+    try:
+        lat = float(request.args.get("lat"))
+        lon = float(request.args.get("lon"))
+        zoom = int(request.args.get("z"))
+    except:
+        return server_error(500)
```

Comment:
Perhaps a 500 error is not correct here - this is really about a wrong or malformed URL